KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+
+The `KCT` layer allows you to interact with Klaytn compatible token (KCT) contracts (e.g. [KIP-7] or [KIP-17]). This is implemented by extending the `Contract` class in the `Contract` layer.
+
+The `Contract` layer allows you to interact with smart contracts on the Klaytn.
+
+The `ABI` layer provides the functions to encode and decode parameters based on ABI.
+
+The `Transaction` layer contains classes of Klaytn transaction types, `TransactionDecoder`, and `TransactionHasher`.
+
+The `TransactionDecoder` class decodes the RLP-encoded transaction string. The `TransactionHasher` class calculates the hash of the transaction.
+
+The `Wallet` layer contains keyring classes (`SingleKeyring`, `MultipleKeyring`, and `RoleBasedKeyring`) and `KeyringContainer` class that contains multiple keyrings. The `KeyringContainer` class acts as an "in-memory wallet" that stores the Keyring instances.
+
+The `Account` layer contains an `Account` class that stores information needed when updating the [AccountKey] of the account in the Klaytn.
+
+Also, the `RPC` layer includes the `Klay` class, which is responsible for rpc calls of the klay namespace, and the `Net` class, which is responsible for rpc calls of the net namespace.
+
+Finally, the `Utils` layer contains the `Utils` class that provides utility functions.
+
+From the next chapter, each layer is described in detail.
+
+### Account Layer Class Diagram
+
+The `Account` layer provides functionality related to updating the [AccountKey] of the Klaytn account.
+
+![0811Account](https://user-images.githubusercontent.com/32922423/89860635-e508c000-dbde-11ea-8c2f-b755212f3556.png)
+
+`Account` is a class that contains information needed to update the [AccountKey] of the account in the Klaytn. It has `address` and `accountKey` as member variables. The `accountKey` can be an instance of (`AccountKeyLegacy`, `AccountKeyPublic`, `AccountKeyFail`, `AccountKeyWeightedMultiSig` or `AccountKeyRoleBased`) depending on the key.
+
+An `Account` instance is injected into the `account` field of the AccountUpdate transaction types (`AccountUpdate`, `FeeDelegatedAccountUpdate` or `FeeDelegatedAccountUpdateWithRatio`). When an AccountUpdate transaction is successfully processed in Klaytn, the account key of the [Klaytn account] is updated with the `accountKey` defined in the `Account` instance.
+
+Interface for AccountKey is defined as `IAccountKey`, and account key classes (`AccountKeyLegacy`, `AccountKeyPublic`, `AccountKeyFail`, `AccountKeyWeightedMultiSig` and `AccountKeyRoleBased`) implements IAccountKey.
+
+`WeightedMultiSigOptions` is provided for `AccountKeyWeightedMultiSig` where the threshold and the weight of each key are defined.
+
+The `AccountKeyDecoder` class decodes the RLP-encoded string using the decode function implemented in each AccountKey class.
+
+#### IAccountKey
+
+`IAccountKey` is the interface of AccountKey. All AccountKey classes must implement `IAccountKey`.
+
+- Variable description
```

Comment:
```suggestion
##### Variable description
```
Please change headers. `-` for a header is not good for visibility.

Reply:
I changed header :)